### PR TITLE
New feature: Unioning source data

### PR DESCRIPTION
**Pull request by pawelngei**
**Are you a current Fivetran customer?** 
Yes, Pawel Ngei from Pattern Brands

**What change(s) does this PR introduce?** 
- Introduce the ability to union data from multiple schemas databases, as per https://github.com/fivetran/dbt_klaviyo_source/pull/9 (for Klaviyo)

**Does this PR introduce a breaking change?**
- [ ] Yes (please provide breaking change details below.)
- [x] No  (please provide explanation how the change is non breaking below.)

**Is this PR in response to a previously created Issue**
- [ ] Yes, Issue
- [x] No 

**How did you test the PR changes?** 
- [ ] CircleCi  
- [x] Other (please provide additional testing details below)

Locally, on Pattern Brands' data, **important**: `adwords` mode only. Needs to be tested on `google_ads`.

**Select which warehouse(s) were used to test the PR**
- [ ] BigQuery
- [ ] Redshift
- [x] Snowflake
- [ ] Postgres
- [ ] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
:mag: 

**Feedback**
Will make it available for review when I create a matching one for Google Ads (not source)

### CODE REVIEW

**Comment by fivetran-joemarkiewicz** [comment]
Thanks so much for opening this PR @pawelngei! 💯 

I see the only item you are waiting on to mark this as "ready for review" is the `google ads` api data. Currently, the Google Ads API is still in private preview and only a few customers have made the transition. Therefore, we could either help you obtain access to a new Google Ads API enabled connector, or we can help tackle the union integration for those models.

I would be more than happy to take on the Google Ads integration while I perform this review during the next sprint. Let me know what your thoughts are!

**Comment by pawelngei** [comment]
> I see the only item you are waiting on to mark this as "ready for review" is the `google ads` api data.

Ah, a slight miscommunication! I just wanted to give you the "front end" plugin, https://github.com/fivetran/dbt_google_ads/ together with this one, not the `google_ads` mode (as opposed to `adwords)! :) You should expect it today!

**Comment by pawelngei** [comment]
I don't know which profile to use to generate the documentation for this plugin, but outside of the docs and some links in the changelog it should be enough! :)

File: models/adwords_connector/tmp/stg_google_ads__click_performance_tmp.sql [hunk 1]
```diff
@@ -1,4 +1,11 @@
...
+{{
+    fivetran_utils.union_data(
+        table_identifier='click_performance',
+        database_variable='google_ads_database', 
+        schema_variable='google_ads_schema', 
+        default_database=target.database,
+        default_schema='google_ads',
+        default_variable='google_ads__click_performance'
+    )
+}}
```
Comment (fivetran-joemarkiewicz) [inline models/adwords_connector/tmp/stg_google_ads__click_performance_tmp.sql]: When building out the social media roll up package we found a way to integrate a custom variable for the union_schema and union_databases if this feature is used across packages in the same dbt_project. Would you be able to add the `union_schema_variable` and `union_database_variable` for these `_tmp` models.

[Here](https://github.com/fivetran/dbt_facebook_pages_source/blob/master/models/tmp/stg_facebook_pages__daily_page_metrics_total_tmp.sql#L1-L12) is an example of how we do this within the dbt_facebook_pages_source package. I would envision the variable names to be `google_ads_union_schemas` and `google_ads_union_databases` respectively.

Let me know if you have any questions on this!

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@pawelngei thanks again so much for opening this PR! We have added a feature to the union_data macro that allows us to define a custom variable for the various sources. It will require adding two lines to the `_tmp` models. Below is an example of the two additional lines:

```sql
        union_schema_variable='google_ads_union_schemas',
        union_database_variable='google_ads_union_databases'
```

Let me know if you have any questions on this request!

File: models/adwords_connector/tmp/stg_google_ads__click_performance_tmp.sql (hunk 1 above)
Comment (pawelngei) [reply models/adwords_connector/tmp/stg_google_ads__click_performance_tmp.sql]: Implemented, tested locally (w/o the social media roll up package)!

**Comment by fivetran-joemarkiewicz** [comment]
Thanks again so much for putting the effort into this PR @pawelngei! There are a few minor changes (tests and formatting updates) that I am going to apply before we include this in a live version of the package. Therefore I am changing the base branch to be a working branch that I will apply these updates to.

Thanks so much!

**Pull request merged**